frontend/src/metabase/timelines/questions/components/TimelineCard/TimelineCard.tsx

```diff
@@ -23,30 +24,40 @@ import {
 export interface TimelineCardProps {
   timeline: Timeline;
   isDefault?: boolean;
-  isVisible?: boolean;
+  visibleEventIds: number[];
   selectedEventIds?: number[];
   onEditEvent?: (event: TimelineEvent) => void;
   onMoveEvent?: (event: TimelineEvent) => void;
   onArchiveEvent?: (event: TimelineEvent) => void;
   onToggleEventSelected?: (event: TimelineEvent, isSelected: boolean) => void;
-  onToggleTimeline?: (timeline: Timeline, isVisible: boolean) => void;
+  onShowTimelineEvents: (timelineEvent: TimelineEvent[]) => void;
+  onHideTimelineEvents: (timelineEvent: TimelineEvent[]) => void;
 }
 
 const TimelineCard = ({
   timeline,
   isDefault,
-  isVisible,
+  visibleEventIds = [],
   selectedEventIds = [],
   onEditEvent,
   onMoveEvent,
   onArchiveEvent,
   onToggleEventSelected,
-  onToggleTimeline,
+  onShowTimelineEvents,
+  onHideTimelineEvents,
 }: TimelineCardProps): JSX.Element => {
   const events = getEvents(timeline.events);
   const isEventSelected = events.some(e => selectedEventIds.includes(e.id));
   const [isExpanded, setIsExpanded] = useState(isDefault || isEventSelected);
 
+  const anyEventVisible = events.some(event =>
```

Comment:
nit: maybe use `useMemo`? seems a bit time consuming

Reply:
And below as well

Reply:
Good one, added.